src/pybind/mgr/dashboard/controllers/grafana.py

```diff
@@ -36,7 +36,14 @@ def validation(self, params):
         method = 'GET'
         url = str(Settings.GRAFANA_API_URL).rstrip('/') + \
             '/api/dashboards/uid/' + params
-        response = grafana.url_validation(method, url)
+        try:
+            response = grafana.url_validation(method, url)
+        except Exception as e:  # pylint: disable=broad-except
+            raise DashboardException(
+                msg=str(e),
+                component='grafana',
+                http_status_code=500,
```

Comment:
Maybe some 4xx error code should be shown?
"A 495 SSL Certificate Error is an expansion of the 400 Bad Request error response code and is used when the client has given an invalid client certificate when making a request."

Reply:
OK, let the DashboardException do the work. The default error code will be 400 then.